template/interpolate/funcs.go

```diff
@@ -327,24 +328,43 @@ func funcGenVault(ctx *Context) interface{} {
 }
 
 func funcGenAwsSecrets(ctx *Context) interface{} {
-	return func(name string) (string, error) {
+	return func(secret ...string) (string, error) {
 		if !ctx.EnableEnv {
 			// The error message doesn't have to be that detailed since
 			// semantic checks should catch this.
 			return "", errors.New("AWS Secrets Manager vars are only allowed in the variables section")
 		}
+
+		// Check if at leas 1 parameter has been used
+		if len(secret) == 0 {
+			return "", errors.New("At least one parameter must be used")
+		}
 		// client uses AWS SDK CredentialChain method. So,credentials can
 		// be loaded from credential file, environment variables, or IAM
 		// roles.
-		client, err := awssmapi.New()
-		if err != nil {
-			return "", fmt.Errorf("Error getting AWS Secrets Manager client: %s", err)
+		client := awssmapi.New(
+			&awssmapi.AWSConfig{},
+		)
```

Comment:
This call no longer returns an error?

Reply:
Correct, this is now just a constructor, errors are being handled in `GetSecret` method.